sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/coders/JAXBCoder.java

```diff
@@ -92,18 +107,42 @@ public T decode(InputStream inStream, Context context) throws CoderException, IO
       }
 
       @SuppressWarnings("unchecked")
-      T obj = (T) jaxbUnmarshaller.unmarshal(new FilterInputStream(inStream) {
-        // JAXB closes the underyling stream so we must filter out those calls.
-        @Override
-        public void close() throws IOException {
-        }
-      });
+      T obj = (T) jaxbUnmarshaller.unmarshal(getInputStream(inStream, context));
       return obj;
     } catch (JAXBException e) {
       throw new CoderException(e);
     }
   }
 
+  private InputStream getInputStream(final InputStream inStream, Context context)
+      throws CoderException, IOException {
+    if (Context.OUTER.equals(context)) {
+      return new FilterInputStream(inStream) {
+        // JAXB closes the underlying stream so we must filter out those calls.
+        @Override
+        public void close() throws IOException {}
```

Comment:
I think this is not safe, because close should flush.

Again, existing bug. *this* fix should be backported.